src/scripts/main.js

```diff
@@ -1,3 +1,280 @@
+  if (activeRow) {
+    activeRow.classList.remove('active');
+  }
+
+  row.classList.add('active');
+
+  activeRow = row;
+}
+// #endregion
+
+// #region form to add new employee
+const form = document.createElement('form');
+
+form.classList.add('new-employee-form');
+
+const fields = [
+  {
+    label: 'Name:',
+    name: 'name',
+    type: 'text',
+    data: 'name',
+  },
+  {
+    label: 'Position:',
+    name: 'position',
+    type: 'text',
+    data: 'position',
+  },
+  {
+    label: 'Age:',
+    name: 'age',
+    type: 'number',
+    data: 'age',
+  },
+  {
+    label: 'Salary:',
+    name: 'salary',
+    type: 'number',
+    data: 'salary',
+  },
+];
+
+const selectField = {
+  label: 'Office:',
+  name: 'office',
+  options: [
+    { value: 'tokyo', text: 'Tokyo' },
+    { value: 'singapore', text: 'Singapore' },
+    { value: 'london', text: 'London' },
+    { value: 'new-York', text: 'New York' },
+    { value: 'edinburgh', text: 'Edinburgh' },
+    { value: 'san-Francisco', text: 'San Francisco' },
+  ],
+  data: 'office',
+};
+
+fields.forEach((field, index) => {
+  const label = document.createElement('label');
+
+  label.textContent = field.label;
+
+  const input = document.createElement('input');
+
+  input.type = field.type;
+  input.name = field.name;
+  input.setAttribute('data-qa', field.data);
+  input.required = true;
+
+  label.appendChild(input);
+  form.appendChild(label);
+
+  if (index === 1) {
+    const selectLabel = document.createElement('label');
+    const select = document.createElement('select');
+
+    selectLabel.textContent = selectField.label;
+    select.name = selectField.name;
+    select.setAttribute('data-qa', selectField.data);
+    // select.required = true;
+
+    selectField.options.forEach((optionData) => {
+      const option = document.createElement('option');
+
+      option.value = optionData.value;
+      option.textContent = optionData.text;
+      select.appendChild(option);
+    });
+
+    selectLabel.appendChild(select);
+    form.appendChild(selectLabel);
+  }
+});
+
+const submitButton = document.createElement('button');
+
+submitButton.type = 'submit';
+submitButton.textContent = 'Save to table';
+form.appendChild(submitButton);
+
+document.body.appendChild(form);
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const formData = new FormData(form);
+  const data = {};
+
+  formData.forEach((value, key) => {
+    data[key] = value;
+  });
+
+  if (validation(data)) {
+    addNewRow(data);
+    pushNotification('Success', 'Row added successfully!', 'success');
+  }
+});
+
+function addNewRow(data) {
+  const tableBody = document.querySelector('tbody');
+  const newRow = document.createElement('tr');
+
+  for (const key in data) {
+    if (data.hasOwnProperty(key)) {
+      const newCell = document.createElement('td');
+
+      newCell.textContent = capitalizeWords(data[key]);
+
+      if (key === 'position') {
+        newCell.textContent = data[key];
+      }
```

Comment:
The condition `key === 'position'` is redundant because the `capitalizeWords` function is already applied to all fields. If specific formatting is needed for the position, consider handling it separately.